consensus/hotstuff/cruisectl/block_rate_controller.go

```diff
@@ -48,6 +39,11 @@ type epochInfo struct {
 	nextEpochFinalView    *uint64
 }
 
+// proposalTimingContainer wraps an instance of ProposalTiming for storing it in atomic.Pointer
+type proposalTimingContainer struct {
```

Comment:
I am a bit confused, why do we need it?

Reply:
good point. Didn't think about just working with a pointer to an interface ... removed `proposalTimingContainer ` in https://github.com/onflow/flow-go/pull/4392/commits/867ccfe3329f0d60463ece3ef67764d0e1d9664b